SU2_CFD/src/solvers/CSpeciesFlameletSolver.cpp

```diff
@@ -0,0 +1,823 @@
+
+    /*--- Add Residual ---*/
+
+    LinSysRes.SubtractBlock(i_point, residual);
+
+    /*--- Implicit part ---*/
+
+    if (implicit) Jacobian.SubtractBlock2Diag(i_point, residual.jacobian_i);
+  }
+  END_SU2_OMP_FOR
+
+  /*--- Axisymmetry source term for the scalar equation. ---*/
+  if (axisymmetric) {
+    CNumerics* numerics = numerics_container[SOURCE_SECOND_TERM + omp_get_thread_num() * MAX_TERMS];
+
+    SU2_OMP_FOR_DYN(omp_chunk_size)
+    for (auto iPoint = 0u; iPoint < nPointDomain; iPoint++) {
+      /*--- Set primitive variables w/o reconstruction ---*/
+
+      numerics->SetPrimitive(solver_container[FLOW_SOL]->GetNodes()->GetPrimitive(iPoint), nullptr);
+
+      /*--- Set scalar variables w/o reconstruction ---*/
+
+      numerics->SetScalarVar(nodes->GetSolution(iPoint), nullptr);
+
+      numerics->SetDiffusionCoeff(nodes->GetDiffusivity(iPoint), 0);
+
+      /*--- Set volume of the dual cell. ---*/
+
+      numerics->SetVolume(geometry->nodes->GetVolume(iPoint));
+
+      /*--- Set y coordinate ---*/
+
+      numerics->SetCoord(geometry->nodes->GetCoord(iPoint), nullptr);
+
+      /*--- Set gradients ---*/
+
+      numerics->SetScalarVarGradient(nodes->GetGradient(iPoint), nullptr);
+
+      auto residual = numerics->ComputeResidual(config);
+
+      /*--- Add Residual ---*/
+
+      LinSysRes.SubtractBlock(iPoint, residual);
+
+      /*--- Implicit part ---*/
+
+      if (implicit) Jacobian.SubtractBlock2Diag(iPoint, residual.jacobian_i);
+    }
+    END_SU2_OMP_FOR
+  }
+}
+
+void CSpeciesFlameletSolver::BC_Inlet(CGeometry* geometry, CSolver** solver_container, CNumerics* conv_numerics,
+                                      CNumerics* visc_numerics, CConfig* config, unsigned short val_marker) {
+  string Marker_Tag = config->GetMarker_All_TagBound(val_marker);
+
+  su2double temp_inlet = config->GetInlet_Ttotal(Marker_Tag);
+  const su2double* inlet_scalar_original = config->GetInlet_SpeciesVal(Marker_Tag);
+  su2double* inlet_scalar = const_cast<su2double*>(inlet_scalar_original);
+
+  CFluidModel* fluid_model_local = solver_container[FLOW_SOL]->GetFluidModel();
+
+  /*--- We compute inlet enthalpy from the temperature and progress variable ---*/
+  su2double enth_inlet = inlet_scalar_original[I_ENTH];
+  fluid_model_local->GetEnthFromTemp(&enth_inlet, inlet_scalar[I_PROGVAR], temp_inlet, inlet_scalar_original[I_ENTH]);
+  inlet_scalar[I_ENTH] = enth_inlet;
+
+  /*--- Loop over all the vertices on this boundary marker ---*/
+
+  SU2_OMP_FOR_STAT(OMP_MIN_SIZE)
+  for (auto iVertex = 0u; iVertex < geometry->nVertex[val_marker]; iVertex++) {
+    auto iPoint = geometry->vertex[val_marker][iVertex]->GetNode();
+
+    /*--- Check if the node belongs to the domain (i.e., not a halo node) ---*/
+
+    if (!geometry->nodes->GetDomain(iPoint)) continue;
+
+    if (config->GetSpecies_StrongBC()) {
+      nodes->SetSolution_Old(iPoint, inlet_scalar);
+
+      LinSysRes.SetBlock_Zero(iPoint);
```

Comment:
Duplication of the base class

Reply:
@pcarruscag For tabulated chemistry we know the inlet temperature T and progress variable C, and we have to compute the inlet enthalpy h from the lookup table using C,T (reverse lookup). We could remove that computation here and do it in SetUniformInlet?

Reply:
OK, I have now done this in the same way as for the source_residual, calling CSpeciesSolver::BC_Inlet from CSpeciesFlameletSolver::BC_Inlet